docs/docs/aztec/writing_efficient_contracts.md

```diff
@@ -0,0 +1,57 @@
+---
+title: Writing Efficient Contracts
+sidebar_position: 2
+tags: [Developers, Contracts]
+---
+
+## Writing functions
+
+On Ethereum L1, all data is public and all execution is completely reproducible. The Aztec L2 takes on the challenge of execution of private functions on private data. This is done client side, along with the generation of corresponding proofs, so that the network can verify the proofs and append any encrypted data/nullifiers (privacy preserving state update).
+
+This highlights a key difference with how public vs private functions are written.
+
+:::info Writing efficiently
+
+- **Public functions** can be written intuitively - optimising for execution/gas as one would for EVM L2s
+- **Private functions** are optimised differently, as they are compiled to a circuit to be proven (see [Thinking in Circuits](https://noir-lang.org/docs/explainers/explainer-writing-noir))
+
+:::
+
+## Assessing efficiency
+
+On Aztec (like other L2s) there are several costs/limit to consider...
+
+- L1 costs - execution, blobs, events
+- L2 costs - public execution, data, logs
+- Local limits - proof generation, execution
+
+### Local Proof generation
+
+Since the proof generation is a significant local burden, being mindful of the gate-count of private functions is important. The gate-count is a proportionate indicator of the memory and time required to prove locally, so should not be ignored.
+
+If private functions are written contrary to the advice in [this section](https://noir-lang.org/docs/explainers/explainer-writing-noir#writing-efficient-noir-for-performant-products), then a program risks hitting local limits.
```

Comment:
If all of the strategies outlined in [this doc](https://docs.google.com/document/d/1wzeBSjwOZ6-QePcnPcst0yhojXg1McvIl7g3B7ilRYw/edit?tab=t.0#heading=h.v6lazgm38yzs) are covered in the noir docs page, then i think just linking to it is sufficient. Although I think it would still be valuable to call out the high level strategies here, without going into the low level detail (e.g. constrained vs unconstrained computation, control flow optimizations, avoiding AND, avoiding comparators, etc)